File: uk-ets-app-registry/uk-ets-registry-api/src/main/java/gov/uk/ets/registry/api/user/admin/shared/UserStatusAction.java
package gov.uk.ets.registry.api.user.admin.shared;

/**
 * Possible user status change actions
 * 
 * @author fragkise
 */
public enum UserStatusAction {

	VALIDATE("Validate user"), UNEROLL("Unenroll user"), SUSPEND("Suspend user"),
	RESTORE("Restore user");

	private String description;

	UserStatusAction(String description) {
		this.description = description;
	}

	public String getDescription() {
		return description;
	}

}

File: uk-ets-app-registry/uk-ets-registry-web/src/app/user-management/model/user-status-action.ts
import { KeycloakUser, UserStatus } from '@shared/user';
import { createAction, props } from '@ngrx/store';
import { ChangeUserStatusActionTypes } from '@user-management/user-details/user-status/store/actions/user-status.actions';

export type UserStatusAction = 'VALIDATE' | 'SUSPEND' | 'RESTORE' | 'UNEROLL';

export interface UserStatusActionOption {
  label: string;
  value: UserStatusAction;
  enabled: true;
  newStatus: UserStatus;
  message?: string;
}

export type UserStatusActionState = Omit<UserStatusActionOption, 'enabled'>;

export interface UserStatusRequest {
  urid: string;
  status: UserStatus;
  comment: string;
}

export interface UserStatusChangedResult {
  userStatus: UserStatus;
  requestId?: string;
  iamIdentifier: string;
  user?: KeycloakUser;
}

export const setComment = createAction(
  ChangeUserStatusActionTypes.SET_COMMENT,
  props<{ comment: string }>()
);
